app/routes.py

```diff
@@ -0,0 +1,413 @@
+def update_existing_customer(customer_id):
+    """Updates and returns details
+    about a specific customer
+    """
+
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return customer_not_found(customer_id), 404
+
+    request_body = request.get_json()
+    if "name" not in request_body or "phone" not in request_body\
+        or "postal_code" not in request_body:
+        return jsonify("Bad Request"), 400
+
+    customer.name = request_body.get("name")
+    customer.postal_code = request_body.get("postal_code")
+    customer.phone = request_body.get("phone")
+    customer.registered_at = NOW
+
+    db.session.commit()
+    response_body = ({
+            "name": f"{customer.name}",
+            "postal_code": f"{customer.postal_code}",
+            "phone": f"{customer.phone}"})
+
+    return response_body, 200
+
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_existing_customer(customer_id):
+    """Deletes a specific customer"""
+
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return customer_not_found(customer_id), 404
+
+    if customer.videos:
+        for rental_record in customer.rentals:
+            db.session.delete(rental_record)
+        db.session.delete(customer)
+        db.session.commit()
+        response_body = {"message" : f"{customer.name} and all their rental records have been deleted"}
+        return jsonify(response_body), 200
+    
+    db.session.delete(customer)
+    db.session.commit()
+
+    return {"id": customer.id}, 200
+
+
+# *****************************
+# *** videos endpoint  CRUD ***
+# *****************************
+@videos_bp.route("", methods = ["GET"])
+def get_videos():
+    """Lists all existing videos
+    and details about each video
+    """
+    videos = Video.query.all()
+    response_body = [video.to_dict() for video in videos]
+
+    return jsonify(response_body), 200
+
+
+@videos_bp.route("/<video_id>", methods = ["GET"])
+def get_video(video_id):
+    """Returns details about
+    a specific video in the store's inventory
+    """
+
+    # check that </video_id> is valid input (ie an id number)
+    if not video_id.isnumeric():
+        return jsonify(None), 400
+    
+    video = Video.query.get(video_id)
+    if video == None:
+        return video_not_found(video_id), 404
+
+    response_body = video.to_dict()
+
+    return jsonify(response_body), 200
+
+
+@videos_bp.route("", methods = ["POST"])
+def post_video():
+    """Creates a new video based on given params"""
+
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return {"details": "Request body must include title."}, 400
+
+    if "release_date" not in request_body:
+        return {"details": "Request body must include release_date."}, 400
+
+    if "total_inventory" not in request_body:
+        return {"details": "Request body must include total_inventory."}, 400
```

Comment:
we can chain these conditionals using the `or` operator as seen in the `PUT` request for `/videos/video_id` below.